### PR TITLE
Kf foods

### PR DESCRIPTION
**Pull request by TheKaterTot**
Creates the /foods endpoint, which will connect to the foods.html on the front end. Fully tested and refactored to include model and controller. 🗡

### CODE REVIEW

File: db/migrations/20170508141720_create-foods-table.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    calories INTEGER,
+    created_at TIMESTAMP
+  )`;
+  return knex.raw(createQuery);
+};
+
+exports.down = function(knex, Promise) {
+  let dropQuery = `DROP TABLE foods`;
+  return knex.raw(dropQuery);
+};
```
Comment (cmacaulay) [inline db/migrations/20170508141720_create-foods-table.js:14]: 🎉 migrations!

File: knexfile.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    connection: process.env.DATABASE_URL,
+    migrations: {
+      directory: './db/migrations'
+    },
+    seeds: {
+      directory: './db/seeds/production'
+    },
+    useNullAsDefault: true
+  }
+};
```
Comment (cmacaulay) [inline knexfile.js:35]: 🎉 knex FTW

File: server.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+const FoodsController = require('./lib/controllers/foods-controller')
+
+app.use(bodyParser.json());
+app.use(bodyParser.urlencoded({ extended: true }))
+
+app.set('port', process.env.PORT || 3000);
+
+app.get('/api/foods', FoodsController.index);
+
+module.exports = app;
```
Comment (cmacaulay) [inline server.js:14]: @TheKaterTot what a lean server, lookin' 😎 🆒 .

File: test/server-test.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+        assert.equal(parsedFood[0].id, id);
+        assert.equal(parsedFood[0].name, name);
+        assert.equal(parsedFood[0].calories, calories);
+        assert.ok(parsedFood[0].created_at);
+        done();
+      })
+    })
+  })
+})
```
Comment (cmacaulay) [inline test/server-test.js:72]: 🎉 testing!
@case-eee in class we were only shown a server.js test, but would you also suggest a model test...and a controller test? What's the best way to break this down?

**Review by cmacaulay — APPROVED** [review]

File: test/server-test.js (hunk 4 above)
Comment (case-eee) [reply test/server-test.js:72]: @cmacaulay YES! test ALL of the things 👍 breaking it down in that way would be great!

File: db/migrations/20170508141720_create-foods-table.js (hunk 1 above)
Comment (case-eee) [reply db/migrations/20170508141720_create-foods-table.js:14]: woooo!

**Comment by case-eee** [comment]
this looks great @TheKaterTot and @cmacaulay 🎉

**Pull request merged**